extractor/main.go

```diff
@@ -105,10 +105,23 @@ func execExtract(ctx context.Context, cfg *extractorCfg) error {
 		return errInvalidOutputDir
 	}
 
-	// Find the files that need to be analyzed
-	sourceFiles, findErr := findFilePaths(cfg.sourceDir, cfg.fileType)
-	if findErr != nil {
-		return fmt.Errorf("unable to find file paths, %w", findErr)
+	var sourceFiles []string
+	var findErr error
+
+	// Check if source is valid*
+	source, err := os.Stat(cfg.sourcePath)
+	if err != nil {
+		return fmt.Errorf("unable to open source, %w", err)
+	}
+	// If source is dir, walk it and add to sourceFiles
+	if source.IsDir() {
+		sourceFiles, findErr = findFilePaths(cfg.sourcePath, cfg.fileType)
+		if findErr != nil {
+			return fmt.Errorf("unable to find file paths, %w", findErr)
+		}
+	} else {
+		// If source is not dir, open the file directly*
+		sourceFiles = append(sourceFiles, cfg.sourcePath)
```

Comment:
You can set the initial `sourceFiles := cfg.sourcePath`, since at the moment of calling this `else` block, the slice is empty

This way, you eliminate the `else` block completely, and redefine the pointer of `sourceFiles` in the `if`

Reply:
`sourceFiles` needs to be a string slice to not break the current API. If I set it to `sourceFiles := cfg.sourcePath`, it will be a string. I made the following simplification, let me know if it is alright: [748ef8d](https://github.com/gnolang/tx-exports/pull/11/commits/748ef8d27458e49eef9309678efdf71324bb52e8)

Reply:
Ah, got it